engine/src/main/java/com/sandboni/core/engine/sta/executor/ExecutorPools.java

```diff
@@ -0,0 +1,22 @@
+package com.sandboni.core.engine.sta.executor;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+public class ExecutorPools {
+
+    // By default we double the number of available cores
+    private static final int THREAD_NUMBER = Runtime.getRuntime().availableProcessors() * 2;
+
+    public static final Executor fixedThreadPool =
+            Executors.newFixedThreadPool(THREAD_NUMBER,
+                    (Runnable r) -> {
+                        Thread thread = new Thread(r);
+                        thread.setDaemon(true);
```

Comment:
why each thread need to run as daemon?

Reply:
To let the JVM finish even if the Thread is blocked for any specific reason. We don't want to hold the whole JVM just for a single worker thread.

Reply:
do we call setDaemon(false) later in the code?

Reply:
No, is not needed and actually not possible after thread creation. We want those threads to be daemon so they die when the JVM shuts down.